parley/src/shape.rs

```diff
@@ -148,12 +186,25 @@ pub fn shape_text<'a, B: Brush>(
             text_range.start = text_range.end;
             char_range.start = char_range.end;
         }
+
+        while let Some(box_idx) = deferred_boxes.pop() {
+            // Push the box to the list of items
+            layout.data.push_inline_box(box_idx);
+        }
```

Comment:
I think this will reverse the order of inline boxes with the same index.

The `deferred_boxes` vec could be replaced by an `Option<Range<usize>>` since they're guaranteed to be contiguous.

Reply:
Good catch! Using a range is also a good idea.

Reply:
Ok. I've updated this to:
- Use `.drain()` to fix the ordering issue
- Hoist the `Vec` allocation out of the loop so it's now only 1 Vec alloc per call to `shape_text`

I'm hoping the range based version can be tackled in a followup.